core/embed/bootloader/messages.c

```diff
@@ -358,8 +358,10 @@ void process_msg_FirmwareErase(uint8_t iface_num, uint32_t msg_size,
 }
 
 static uint32_t chunk_size = 0;
-// SRAM is unused, so we can use it for chunk buffer
-uint8_t *const chunk_buffer = (uint8_t *const)0x20000000;
+
+__attribute__((section(".buf"))) uint32_t chunk_buffer[IMAGE_CHUNK_SIZE / 4];
+
+#define CHUNK_BUFFER_PTR (const uint8_t *const) & chunk_buffer
```

Comment:
Let's add brackets to be 100% sure this won't explode badly when used in some funky context.

```suggestion
#define CHUNK_BUFFER_PTR ((const uint8_t *const) & chunk_buffer)
```

Reply:
fixed 8b066418a63c9c7989052501a17975e85c7e28d4